Month3/LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,334 @@
+#pragma once
+# include <iostream>
+#include <string>
+#include <functional>
+template <typename T>
+class TSLList
+{
+private:
+	
+	struct FNode
+	{
+		T Element;
+		FNode* Next = nullptr ;
+	};
+	
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+	
+	int Size = 0;
+public:
+	TSLList()
+	{
+
+	}
+	TSLList(const T& NewElement)
+	{
+		Head = new FNode();
+		Head->Element = NewElement;
+		Tail = Head;
+		Size = 1;
+	}
+
+	TSLList(const TSLList<T>& OtherList)
+	{
+		FNode* Current = OtherList.Head;
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	TSLList& operator=(const TSLList<T>& OtherList)
+	{
+		Clear();
+		FNode* Current = OtherList.Head;
+		for (int CurrentIndex = 0; CurrentIndex < OtherList.Size; CurrentIndex++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+		return *this;
+	}
+
```

Comment:
well done, but i think there si no need of the return, this could be void.

Reply:
jaja please check your writing, si(is) no need, and  in my case I believe in this case is better return